shared/src/main/java/org/odk/collect/shared/files/FileExt.kt

```diff
@@ -0,0 +1,60 @@
+package org.odk.collect.shared.files
+
+import java.io.File
+import java.io.IOException
+import java.io.InputStream
+import kotlin.jvm.Throws
+
+object FileExt {
+    /**
+     * Original File.getCanonicalPath() may return paths with inconsistent letter casing for the
+     * /Android/data/ part of the path, such as /storage/emulated/0/android/data/... or /storage/emulated/0/Android/Data/...
+     * instead of the expected /storage/emulated/0/Android/data/...
+     * Since the Android file system is case-sensitive, this behavior appears to be a bug.
+     *
+     * For more details, see the discussion on Stack Overflow:
+     * https://stackoverflow.com/questions/78965720/file-getcanonicalpath-returns-inconsistent-letter-casing-in-path
+     */
+    fun File.sanitizedCanonicalPath(): String {
+        val androidDataSegment = "/Android/data/"
+
+        val canonicalPath = canonicalPath
+
+        if (canonicalPath.contains(androidDataSegment, true)) {
+            val regex = Regex(androidDataSegment, RegexOption.IGNORE_CASE)
+            return canonicalPath.replace(regex, androidDataSegment)
```

Comment:
I think you should be able to do the replace without a contains check -- if there's no match it just won't be replaced.

Reply:
Good point.